src/replication/adapter/aliacr/adapter_test.go

```diff
@@ -0,0 +1,275 @@
+			AccessSecret: "MockAccessSecret",
+		},
+		URL: "https://registry.test-region.aliyuncs.com",
+	})
+	assert.Nil(t, err)
+	assert.NotNil(t, adapter)
+
+	// test case for URL is cr service.
+	adapter, err = factory(&model.Registry{
+		Type: model.RegistryTypeAliAcr,
+		Credential: &model.Credential{
+			AccessKey:    "MockAccessKey",
+			AccessSecret: "MockAccessSecret",
+		},
+		URL: "https://cr.test-region.aliyuncs.com",
+	})
+	assert.Nil(t, err)
+	assert.NotNil(t, adapter)
+
+}
+
+func getMockAdapter(t *testing.T, hasCred, health bool) (*adapter, *httptest.Server) {
+	server := test.NewServer(
+		// &test.RequestHandlerMapping{
+		// 	Method:  http.MethodGet,
+		// 	Pattern: "/v2/_catalog",
+		// 	Handler: func(w http.ResponseWriter, r *http.Request) {
+		// 		w.WriteHeader(http.StatusOK)
+		// 		w.Write([]byte(`
+		// {
+		// 	"repositories": [
+		// 			"test1"
+		// 	]
+		// }`))
+		// 	},
+		// },
+		// &test.RequestHandlerMapping{
+		// 	Method:  http.MethodGet,
+		// 	Pattern: "/v2/{repo}/tags/list",
+		// 	Handler: func(w http.ResponseWriter, r *http.Request) {
+		// 		w.WriteHeader(http.StatusOK)
+		// 		w.Write([]byte(`
+		// 	{
+		// 	    "name": "test1",
+		// 	    "tags": [
+		// 	        "latest"
+		// 	    ]
+		// 	}`))
+		// 	},
+		// },
+		&test.RequestHandlerMapping{
+			Method:  http.MethodGet,
+			Pattern: "/v2/",
+			Handler: func(w http.ResponseWriter, r *http.Request) {
+				fmt.Println(r.Method, r.URL)
+				if health {
+					w.WriteHeader(http.StatusOK)
+				} else {
+					w.WriteHeader(http.StatusBadRequest)
+				}
+			},
+		},
+		&test.RequestHandlerMapping{
+			Method:  http.MethodGet,
+			Pattern: "/",
+			Handler: func(w http.ResponseWriter, r *http.Request) {
+				fmt.Println(r.Method, r.URL)
+				w.WriteHeader(http.StatusOK)
+			},
+		},
+		&test.RequestHandlerMapping{
+			Method:  http.MethodPost,
+			Pattern: "/",
+			Handler: func(w http.ResponseWriter, r *http.Request) {
+				fmt.Println(r.Method, r.URL)
+				if buf, e := ioutil.ReadAll(&io.LimitedReader{R: r.Body, N: 80}); e == nil {
+					fmt.Println("\t", string(buf))
+				}
+				w.WriteHeader(http.StatusOK)
+			},
+		},
+	)
+
+	registry := &model.Registry{
+		Type: model.RegistryTypeAliAcr,
+		URL:  server.URL,
+	}
+	if hasCred {
+		registry.Credential = &model.Credential{
+			AccessKey:    "MockAccessKey",
+			AccessSecret: "MockAccessSecret",
+		}
+	}
+	nativeRegistry, err := native.NewAdapter(registry)
+	if err != nil {
+		panic(err)
+	}
+	return &adapter{
+		Adapter:  nativeRegistry,
+		region:   "test-region",
+		domain:   server.URL,
+		registry: registry,
+	}, server
+}
+
+func TestAdapter_Info(t *testing.T) {
+	a, s := getMockAdapter(t, true, true)
+	defer s.Close()
+	info, err := a.Info()
+	assert.Nil(t, err)
+	assert.NotNil(t, info)
+
+	assert.EqualValues(t, 1, len(info.SupportedResourceTypes))
+	assert.EqualValues(t, model.ResourceTypeImage, info.SupportedResourceTypes[0])
+}
+
+func TestAdapter_HealthCheck(t *testing.T) {
```

Comment:
Remove it and `Test_adapter_PrepareForPush` as the two methods have been removed from the adapter